packages/dds/task-manager/src/interfaces.ts

```diff
@@ -47,6 +49,32 @@ export interface ITaskManager extends ISharedObject<ITaskManagerEvents> {
     queued(taskId: string): boolean;
 }
 
+export interface IOldestClientObservableEvents extends IEvent {
+    (event: "connected", listener: () => void);
+    // Typescript won't convert IFluidDataStoreRuntime and ContainerRuntime if we unify these,
+    // I believe this is because the "connected" event has a clientId arg in the runtimes.
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    (event: "disconnected", listener: () => void);
+}
+
+/**
+ * This is to make OldestClientObserver work with either a ContainerRuntime or an IFluidDataStoreRuntime
+ * (both expose the relevant API surface and eventing).  However, really this info probably shouldn't live on either,
+ * since neither is really the source of truth (they are just the only currently-available plumbing options).
+ * It's information about the connection, so the real source of truth is lower (at the connection layer).
+ */
+export interface IOldestClientObservable extends IEventProvider<IOldestClientObservableEvents> {
```

Comment:
I chose "observable" since it's an object which provides sufficient API surface to observe whether we are the oldest client.

I'm open to alternatives, though I prefer "observable/observer" over "port/adapter" since the input/output directionality is hinted at (whoever consumes an observable is doing the observing, whoever implements observable is exposing information to be observed), and similar preference over "dependency" since it hints at the nature of the dependency.